### DISCUSSION TITLE
query table partition data table alias issue

### DISCUSSION THREAD

**Discussion by travelliu** [dbeaver/dbeaver#33199] [DBeaver Development]
### Description

SELECT x.* FROM public.newtable_1 PARTITION(part_1) x --> correct

SELECT x.* FROM public.newtable_1 x PARTITION(part_1)  error

so

https://github.com/dbeaver/dbeaver/blob/107958e9d1f3e1b181c554d472a16c20cb1dc6e3/plugins/org.jkiss.dbeaver.model.jdbc/src/org/jkiss/dbeaver/model/impl/jdbc/struct/JDBCTable.java#L157C1-L166C78

```
        StringBuilder query = new StringBuilder(100);
        query.append("SELECT ");
        appendSelectSource(monitor, query, tableAlias, rowIdAttribute);
        query.append(" FROM ").append(getTableName());
        if (tableAlias != null) {
            query.append(" ").append(tableAlias); //$NON-NLS-1$
        }
        appendExtraSelectParameters(query);
        SQLUtils.appendQueryConditions(dataSource, query, tableAlias, dataFilter);
        SQLUtils.appendQueryOrder(dataSource, query, tableAlias, dataFilter);
```
need change to 

```
        appendExtraSelectParameters(query);
     if (tableAlias != null) {
            query.append(" ").append(tableAlias); //$NON-NLS-1$
        }
```

### DBeaver Version

23.3.0

**Comment by LonwoLonwo** [comment]
Hello @travelliu 

Sorry, what you are doing is not clear.
Please provide clear steps to reproduce your issue.

**Comment by E1izabeth** [comment]
@travelliu could you share the way to reproduce the issue?

**Comment by travelliu** [comment]
> @travelliu could you share the way to reproduce the issue?

No. I just see that the SQL statement generated by this code is not logically correct.
Oracle aliases should be after the partition name.

appendExtraSelectParameters Functions now seem to just add partition syntax

**Comment by E1izabeth** [comment]
@hallstorm it's not connected to this issue

**Comment by E1izabeth** [comment]
We don't add aliases when building select from partition
https://github.com/dbeaver/dbeaver/blob/107958e9d1f3e1b181c554d472a16c20cb1dc6e3/plugins/org.jkiss.dbeaver.ext.oracle/src/org/jkiss/dbeaver/ext/oracle/model/OracleTablePartition.java#L257-L263

![image](https://github.com/dbeaver/dbeaver/assets/28875055/4caa8248-0c2a-4aee-9162-518efd339301)
